friends/src/FriendsList.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import axios from 'axios';
+import Friend from './Friend';
+import UpdateFriend from './UpdateFriend';
+import FriendListHeader from './FriendListHeader';
+import { ListWrapper, StyledForm } from './Styled';
+
+class FriendsList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      friends: [],
+      name: '',
+      age: '',
+      email: '',
+      id: ''
+    };
+  }
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:5000/friends')
+      .then(data => {
+        // console.log(data.data);
+        this.setState({ friends: data.data });
+        // console.log(this.state);
+      })
+      .catch(() => alert('GET Error'));
+  }
+
+  handleInput = event => {
+    event.preventDefault();
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+    // console.log(this.state.friends);
+    if (this.state.name && this.state.age && this.state.email) {
+      const name = this.state.name;
+      const age = +this.state.age;
+      const email = this.state.email;
+
+      axios
+        .post('http://localhost:5000/friends', { name, age, email })
+        .then(res => {
+          //   console.log(res);
+          //   console.log(res.data);
+          this.setState({
+            friends: res.data,
+            name: '',
+            age: '',
+            email: ''
+          });
+        })
+        .catch(() => alert('POST Error'));
+    } else {
+      alert('Please enter name, age and email');
```

Comment:
Great job working in some helpful error messages, and thinking about cases where a user may try and submit incomplete form data w/line #39 👍